File: aroundHubSpringBoot/src/main/java/com/example/aroungHubJavaSpringBoot/aroundHubSpringBoot/data/dao/impl/ProductDAOImpl.java
package com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.dao.impl;

import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.dao.ProductDAO;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.entity.ProductEntity;
import com.example.aroungHubJavaSpringBoot.aroundHubSpringBoot.data.repository.ProductRepository;
import org.springframework.beans.factory.annotation.Autowired;

public class ProductDAOImpl implements ProductDAO {

    ProductRepository productRepository;

    /*
    * 자동으로 연결시켜 준다.
    * Dependency Injection
    */
    @Autowired
    public ProductDAOImpl(ProductRepository productRepository){
        this.productRepository = productRepository;
    }

    @Override
    public ProductEntity saveProduct(ProductEntity productEntity) {
        productRepository.save(productEntity);
        return productEntity;
    }

    @Override
    public ProductEntity getProduct(String productId) {
        ProductEntity productEntity = productRepository.getById(productId);
        return productEntity;
    }
}

File: aroundHubKotlinSpringBoot/src/main/kotlin/com/example/aroungHubKotlinSpringBoot/aroundHubKotlinSpringBoot/data/dao/impl/ProductDAOImpl.kt
package com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.dao.impl

import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.dao.ProductDAO
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.entity.ProductEntity
import com.example.aroungHubKotlinSpringBoot.aroundHubKotlinSpringBoot.data.repository.ProductRepository
import org.springframework.stereotype.Service

@Service
class ProductDAOImpl(
    val productRepository: ProductRepository
) : ProductDAO
{
    override fun saveProduct(productEntity: ProductEntity): ProductEntity {
        productRepository.save(productEntity)
        return productEntity
    }

    override fun getProduct(productId: String): ProductEntity {
        return productRepository.getReferenceById(productId)
    }
}